File: tests/sdk/python/helm-local/step1/__main__.py
#  Copyright 2016-2021, Pulumi Corporation.
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.

import pulumi
from pulumi_kubernetes.core.v1 import ConfigMap, ConfigMapInitArgs
from pulumi_kubernetes.helm.v2 import Chart, LocalChartOpts
from pulumi_kubernetes.helm.v3 import Release, ReleaseArgs

values = {"service": {"type": "ClusterIP"}}

chart = Chart("nginx", LocalChartOpts(path="nginx-chart", values=values))
foo = ConfigMap("foo", ConfigMapInitArgs(data={"foo": "bar"}), opts=pulumi.ResourceOptions(depends_on=chart.ready))

# Deploy a duplicate chart with a different resource prefix to verify that multiple instances of the Chart
# can be managed in the same stack.
Chart("nginx", LocalChartOpts(path="nginx-chart", resource_prefix="dup", values=values))

# Deploy the same chart but using v3.Release with local path argument
Release('nginx-release', ReleaseArgs(chart="nginx", path="nginx-chart", values=values))

File: tests/sdk/go/helm-local/step1/main.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"

	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/helm/v3"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		chart, err := helm.NewChart(ctx, "test", helm.ChartArgs{
			Path:           pulumi.String("nginx"),
			Values:         pulumi.Map{"service": pulumi.StringMap{"type": pulumi.String("ClusterIP")}},
			ResourcePrefix: "prefix",
		})
		if err != nil {
			return err
		}

		svc := chart.GetResource("v1/Service", "prefix-prefix-test-nginx", "").
			ApplyT(func(r any) (any, error) {
				svc := r.(*corev1.Service)
				return svc.Spec.ClusterIP(), nil
			})
		ctx.Export("svc_ip", svc)

		_, err = corev1.NewConfigMap(ctx, "cm", &corev1.ConfigMapArgs{
			Data: pulumi.StringMap{
				"foo": pulumi.String("bar"),
			},
		}, pulumi.DependsOnInputs(chart.Ready))
		if err != nil {
			return err
		}

		return nil
	})
}
